File: src/tauri/commands.ts
/**
 * TypeScript bindings for Tauri backend commands
 */
import { invoke } from '@tauri-apps/api/core';

export interface SSHConnectionInfo {
  host: string;
  port: number;
  username: string;
  credential: string;
}

export interface VNCConnectionInfo {
  host: string;
  port: number;
  password?: string;
}

export interface SFTPConnectionInfo {
  host: string;
  port: number;
  username: string;
  credential: string;
}

/**
 * Open Pi Connect auth webview and capture credentials
 * @returns Auth token/session cookie
 */
export async function openAuthWebview(): Promise<string> {
  return await invoke<string>('open_auth_webview');
}

/**
 * Open system SSH terminal with connection info
 */
export async function openSSHTerminal(info: SSHConnectionInfo): Promise<void> {
  await invoke('open_ssh_terminal', {
    host: info.host,
    port: info.port,
    username: info.username,
    credential: info.credential,
  });
}

/**
 * Open VNC client with connection info
 */
export async function openVNCClient(info: VNCConnectionInfo): Promise<void> {
  await invoke('open_vnc_client', {
    host: info.host,
    port: info.port,
    password: info.password,
  });
}

/**
 * Open SFTP client with connection info
 */
export async function openSFTPClient(info: SFTPConnectionInfo): Promise<void> {
  await invoke('open_sftp_client', {
    host: info.host,
    port: info.port,
    username: info.username,
    credential: info.credential,
  });
}

File: src-tauri/src/commands.rs
use tauri::Manager;

/// Open Pi Connect login page in a webview window
/// Returns captured auth token/cookies
#[tauri::command]
pub async fn open_auth_webview(app_handle: tauri::AppHandle) -> Result<String, String> {
    // TODO: Create new webview window pointing to connect.raspberrypi.com
    // Monitor for successful login (URL change, cookies, etc.)
    // Extract and return auth token
    
    // For now, return error indicating not implemented
    Err("Auth webview not yet implemented".to_string())
}

/// Open SSH terminal for given connection info
#[tauri::command]
pub async fn open_ssh_terminal(
    host: String,
    port: u16,
    username: String,
    credential: String,
) -> Result<(), String> {
    // TODO: Launch system terminal with SSH command
    // Linux: x-terminal-emulator or gnome-terminal
    // macOS: Terminal.app or iTerm2
    // Windows: Windows Terminal or cmd.exe
    
    #[cfg(target_os = "linux")]
    {
        let ssh_cmd = format!("ssh -p {} {}@{}", port, username, host);
        // TODO: Launch terminal with ssh_cmd
        return Err(format!("SSH terminal not implemented: {}", ssh_cmd));
    }
    
    #[cfg(target_os = "macos")]
    {
        return Err("SSH terminal not implemented for macOS".to_string());
    }
    
    #[cfg(target_os = "windows")]
    {
        return Err("SSH terminal not implemented for Windows".to_string());
    }
    
    #[allow(unreachable_code)]
    Err("Unsupported platform".to_string())
}

/// Open VNC client for given connection info
#[tauri::command]
pub async fn open_vnc_client(
    host: String,
    port: u16,
    password: Option<String>,
) -> Result<(), String> {
    // TODO: Launch VNC client
    // Linux: vncviewer, vinagre, remmina
    // macOS: Screen Sharing.app
    // Windows: TightVNC, RealVNC
    
    let vnc_url = if let Some(_pwd) = password {
        format!("vnc://{}:{}", host, port)
    } else {
        format!("vnc://{}:{}", host, port)
    };
    
    Err(format!("VNC client not implemented: {}", vnc_url))
}

/// Open SFTP client for given connection info
#[tauri::command]
pub async fn open_sftp_client(
    host: String,
    port: u16,
    username: String,
    credential: String,
) -> Result<(), String> {
    // TODO: Launch SFTP client or file manager
    // Linux: nautilus, dolphin, thunar with sftp:// URL
    // macOS: Finder with sftp:// URL
    // Windows: WinSCP, FileZilla
    
    let sftp_url = format!("sftp://{}@{}:{}", username, host, port);
    Err(format!("SFTP client not implemented: {}", sftp_url))
}
